GDJS/GDJS/Extensions/Builtin/AsyncExtension.cpp

```diff
@@ -44,19 +44,26 @@ AsyncExtension::AsyncExtension() {
             parentContext.IsInsideAsync()
                 ? "const parentAsyncObjectsList = asyncObjectsList;\n"
                 : "";
-        gd::String asyncObjectsListBuilder =
+        gd::String asyncContextBuilder =
             parentContext.IsInsideAsync()
                 ? "const asyncObjectsList = "
                   "gdjs.LongLivedObjectsList.from(parentAsyncObjectsList);\n"
                 : "const asyncObjectsList = new gdjs.LongLivedObjectsList();\n";
+
+        if (codeGenerator.HasProjectAndLayout()) {
+          asyncContextBuilder +=
+              "asyncObjectsList.backupLocalVariablesContainers(" +
+              codeGenerator.GetCodeNamespace() + ".localVariables);\n";
+        }
```

Comment:
```
else {
 asyncContextbuilder += `// Unsupported local variables in functions.`
}
```

(exact same comment as before so that we can search all occurrences of this)